Make docs fixed width

bfirsh opened the pull request:
When the docs are viewed on wide browser windows, the text becomes hard to read. On large screens, it's ridiculously wide and hard to read.

I have made the docs fixed width like the rest of the website so the text is always a readable width.

I have also generally cleaned up the menus and fixed some bugs that are particularly noticeable at the fixed width:

1. Cleaned up left hand menu by fixing the arrow getting pushed off by long text.

    - Before:

       ![screenshot 2016-07-29 11 53 47](https://cloud.githubusercontent.com/assets/40906/17247207/97f7762e-5589-11e6-9900-64850971eb3e.png)

    - After:

       ![screenshot 2016-07-29 11 53 56](https://cloud.githubusercontent.com/assets/40906/17247215/a170f356-5589-11e6-9435-a9cb9f70d3bf.png)

2. Removed the top-level link in the table of contents, stopped the links from falling off the right hand page, and removed some unnecessary padding.

    - Before:

       ![screenshot 2016-07-29 11 59 50](https://cloud.githubusercontent.com/assets/40906/17247271/072aacbe-558a-11e6-9084-919d0eb0a2f6.png)

    - After:

       ![screenshot 2016-07-29 12 07 40](https://cloud.githubusercontent.com/assets/40906/17247279/13486798-558a-11e6-8174-0d2274d4415f.png)

(Based on top of #297. Intentionally separate so this diff would be easier to read, but that isn't merged yet. In the meantime, view the separate commits to see what changes have been made.)

/cc @FSHA @londoncalling @SvenDowideit

londoncalling commented:
Hi Ben,
Thanks for the ideas on the docs, especially fixing the right-side list of
topics so they don't run off the page :-)

Fisher and Laura have been working on the docs layout, themes, and so
forth. Lightening up the left-side nav bar was actually a purposeful change
to make the UI cleaner, and provide less distraction. They are still
working on making that nav bar more readable/usable without bringing back
so much color.

It would be great if you could synch up with the docs team about your
ideas. I'm cc'ing folks here to bring them into the loop, and see if we
want to meet up or discuss further here.

Thanks,
Vicky

On Fri, Jul 29, 2016 at 4:46 AM, Ben Firshman <notifications@github.com>
wrote:

> When the docs are viewed on wide browser windows, the text becomes hard to
> read. On large screens, it's ridiculously wide and hard to read.
>
> I have made the docs fixed width like the rest of the website so the text
> is always a readable width.
>
> I have also generally cleaned up the menus and fixed some bugs that are
> particularly noticeable at the fixed width:
>
>    1.
>
>    Cleaned up left hand menu by fixing the arrow getting pushed off by
>    long text.
>    -
>
>       Before:
>
>       [image: screenshot 2016-07-29 11 53 47]
>       <https://cloud.githubusercontent.com/assets/40906/17247207/97f7762e-5589-11e6-9900-64850971eb3e.png>
>       -
>
>       After:
>
>       [image: screenshot 2016-07-29 11 53 56]
>       <https://cloud.githubusercontent.com/assets/40906/17247215/a170f356-5589-11e6-9435-a9cb9f70d3bf.png>
>       2.
>
>    Removed the top-level link in the table of contents, stopped the links
>    from falling off the right hand page, and removed some unnecessary padding.
>    -
>
>       Before:
>
>       [image: screenshot 2016-07-29 11 59 50]
>       <https://cloud.githubusercontent.com/assets/40906/17247271/072aacbe-558a-11e6-9084-919d0eb0a2f6.png>
>       -
>
>       After:
>
>       [image: screenshot 2016-07-29 12 07 40]
>       <https://cloud.githubusercontent.com/assets/40906/17247279/13486798-558a-11e6-8174-0d2274d4415f.png>
>
> (Based on top of #297 <https://github.com/docker/docs-base/pull/297>.
> Intentionally separate so this diff would be easier to read, but that isn't
> merged yet. In the meantime, view the separate commits to see what changes
> have been made.)
>
> /cc @FSHA <https://github.com/FSHA> @londoncalling
> <https://github.com/londoncalling> @SvenDowideit
> <https://github.com/SvenDowideit>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/docker/docs-base/pull/298
> Commit Summary
>
>    - Switch from DOS line endings to Unix
>    - Make containers fixed width
>    - Fix arrow getting pushed aside by text
>    - Fix outer table of contents margin
>    - Fix table of contents links falling off right
>    - Remove page title from table of contents
>    - Shorten text in search box so it fits
>
> File Changes
>
>    - *M* themes/docker-2016/layouts/404.html
>    <https://github.com/docker/docs-base/pull/298/files#diff-0> (2)
>    - *M* themes/docker-2016/layouts/_default/single.html
>    <https://github.com/docker/docs-base/pull/298/files#diff-1> (4)
>    - *M* themes/docker-2016/layouts/index.html
>    <https://github.com/docker/docs-base/pull/298/files#diff-2> (2)
>    - *M* themes/docker-2016/layouts/partials/hero.html
>    <https://github.com/docker/docs-base/pull/298/files#diff-3> (28)
>    - *M* themes/docker-2016/layouts/partials/heronav.html
>    <https://github.com/docker/docs-base/pull/298/files#diff-4> (24)
>    - *M* themes/docker-2016/layouts/partials/navbar-inner.html
>    <https://github.com/docker/docs-base/pull/298/files#diff-5> (31)
>    - *M* themes/docker-2016/static/documentation.css
>    <https://github.com/docker/docs-base/pull/298/files#diff-6> (652)
>    - *M* themes/docker-2016/static/menu.js
>    <https://github.com/docker/docs-base/pull/298/files#diff-7> (28)
>
> Patch Links:
>
>    - https://github.com/docker/docs-base/pull/298.patch
>    - https://github.com/docker/docs-base/pull/298.diff
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/docker/docs-base/pull/298>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALHuA1HcnTWzLrq-AlAC5pB_2nAvuGItks5qaef_gaJpZM4JYHgt>
> .
>

bfirsh commented:
Thanks Vicky!

FWIW, I haven't changed anything the left hand bar apart from the bug with
the text and arrow overlapping.

On Fri, Jul 29, 2016 at 3:30 PM, Victoria <notifications@github.com> wrote:

> Hi Ben,
> Thanks for the ideas on the docs, especially fixing the right-side list of
> topics so they don't run off the page :-)
>
> Fisher and Laura have been working on the docs layout, themes, and so
> forth. Lightening up the left-side nav bar was actually a purposeful change
> to make the UI cleaner, and provide less distraction. They are still
> working on making that nav bar more readable/usable without bringing back
> so much color.
>
> It would be great if you could synch up with the docs team about your
> ideas. I'm cc'ing folks here to bring them into the loop, and see if we
> want to meet up or discuss further here.
>
> Thanks,
> Vicky
>
> On Fri, Jul 29, 2016 at 4:46 AM, Ben Firshman <notifications@github.com>
> wrote:
>
> > When the docs are viewed on wide browser windows, the text becomes hard
> to
> > read. On large screens, it's ridiculously wide and hard to read.
> >
> > I have made the docs fixed width like the rest of the website so the text
> > is always a readable width.
> >
> > I have also generally cleaned up the menus and fixed some bugs that are
> > particularly noticeable at the fixed width:
> >
> > 1.
> >
> > Cleaned up left hand menu by fixing the arrow getting pushed off by
> > long text.
> > -
> >
> > Before:
> >
> > [image: screenshot 2016-07-29 11 53 47]
> > <
> https://cloud.githubusercontent.com/assets/40906/17247207/97f7762e-5589-11e6-9900-64850971eb3e.png
> >
> > -
> >
> > After:
> >
> > [image: screenshot 2016-07-29 11 53 56]
> > <
> https://cloud.githubusercontent.com/assets/40906/17247215/a170f356-5589-11e6-9435-a9cb9f70d3bf.png
> >
> > 2.
> >
> > Removed the top-level link in the table of contents, stopped the links
> > from falling off the right hand page, and removed some unnecessary
> padding.
> > -
> >
> > Before:
> >
> > [image: screenshot 2016-07-29 11 59 50]
> > <
> https://cloud.githubusercontent.com/assets/40906/17247271/072aacbe-558a-11e6-9084-919d0eb0a2f6.png
> >
> > -
> >
> > After:
> >
> > [image: screenshot 2016-07-29 12 07 40]
> > <
> https://cloud.githubusercontent.com/assets/40906/17247279/13486798-558a-11e6-8174-0d2274d4415f.png
> >
> >
> > (Based on top of #297 <https://github.com/docker/docs-base/pull/297>.
> > Intentionally separate so this diff would be easier to read, but that
> isn't
> > merged yet. In the meantime, view the separate commits to see what
> changes
> > have been made.)
> >
> > /cc @FSHA <https://github.com/FSHA> @londoncalling
> > <https://github.com/londoncalling> @SvenDowideit
> > <https://github.com/SvenDowideit>
> > ------------------------------
> > You can view, comment on, or merge this pull request online at:
> >
> > https://github.com/docker/docs-base/pull/298
> > Commit Summary
> >
> > - Switch from DOS line endings to Unix
> > - Make containers fixed width
> > - Fix arrow getting pushed aside by text
> > - Fix outer table of contents margin
> > - Fix table of contents links falling off right
> > - Remove page title from table of contents
> > - Shorten text in search box so it fits
> >
> > File Changes
> >
> > - *M* themes/docker-2016/layouts/404.html
> > <https://github.com/docker/docs-base/pull/298/files#diff-0> (2)
> > - *M* themes/docker-2016/layouts/_default/single.html
> > <https://github.com/docker/docs-base/pull/298/files#diff-1> (4)
> > - *M* themes/docker-2016/layouts/index.html
> > <https://github.com/docker/docs-base/pull/298/files#diff-2> (2)
> > - *M* themes/docker-2016/layouts/partials/hero.html
> > <https://github.com/docker/docs-base/pull/298/files#diff-3> (28)
> > - *M* themes/docker-2016/layouts/partials/heronav.html
> > <https://github.com/docker/docs-base/pull/298/files#diff-4> (24)
> > - *M* themes/docker-2016/layouts/partials/navbar-inner.html
> > <https://github.com/docker/docs-base/pull/298/files#diff-5> (31)
> > - *M* themes/docker-2016/static/documentation.css
> > <https://github.com/docker/docs-base/pull/298/files#diff-6> (652)
> > - *M* themes/docker-2016/static/menu.js
> > <https://github.com/docker/docs-base/pull/298/files#diff-7> (28)
> >
> > Patch Links:
> >
> > - https://github.com/docker/docs-base/pull/298.patch
> > - https://github.com/docker/docs-base/pull/298.diff
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/docker/docs-base/pull/298>, or mute the thread
> > <
> https://github.com/notifications/unsubscribe-auth/ALHuA1HcnTWzLrq-AlAC5pB_2nAvuGItks5qaef_gaJpZM4JYHgt
> >
> > .
> >
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/docker/docs-base/pull/298#issuecomment-236195693>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AACfyg-Zx-Tr_j-xXesTslPsO13apGVUks5qag51gaJpZM4JYHgt>
> .
>

SvenDowideit commented:
👍  especially to the overlapping bug fix (maybe worth bringing that into a separate PR we can merge straight away).

as per discussion in slack - I find the width you've set way too narrow - not all readers are the same.

we definitely need something like this for users that run maximised - especially on the mega widescreens we have these days - but fixed width isn't allowing users like me to get enough width to suite the way we read either.

(I'm hoping that @FSHA has mad UX skilz to help different readers get what they need)

bfirsh commented:
I have pulled out the bug fixes into a separate pull request so we can do this separately. :)

#300
